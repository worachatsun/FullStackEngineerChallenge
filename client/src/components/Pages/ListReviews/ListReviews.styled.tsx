import styled from 'styled-components';

export const FlexAlignCenter = styled.div`
  display: flex;
  align-items: center;
`;

export const Container = styled(FlexAlignCenter)`
  width: 100%;
  flex-direction: column;
`;

export const Wrapper = styled(FlexAlignCenter)`
  width: 70%;
  flex-direction: column;
`;

export const ListHeader = styled(FlexAlignCenter)`
  flex-direction: row;
  width: 100%;
  justify-content: space-between;
  margin: 0 0 30px 0;
`;

export const Table = styled.div`
  text-align: left;
  width: 100%;
  padding: 20px;
  box-shadow: 2px 0px 4px 0px rgba(0, 0, 0, 0.1);
`;

export const Th = styled.div`
  font-size: 1.1em;
  display: flex;
`;

export const Tr = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-top: 10px;
  height: 40px;
  border: 1px solid #ff0033;
  border-radius: 5px;
  color: #ff0033;

  &:hover {
    color: white;
    background: #ff0033;
  }
`;

export const ButtonContainer = styled.div`
  display: flex;
  justify-content: space-around;
  width: 30%;
`;

export const Hr = styled.hr`
  border: 1px solid #444140;
  opacity: 0.5;
`;

export const Lable = styled.div`
  font-size: 1.5em;
`;

export const TrText = styled.div`
  margin-left: 10px;
`;
