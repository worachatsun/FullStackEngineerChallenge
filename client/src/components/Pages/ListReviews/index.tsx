import React, { FunctionComponent, useContext, useState } from "react";
import useSWR from "swr";
import { REVIEW_LIST_API } from "../../../constants/routes";
import { UserContext } from "../../../context/UserContext";
import Modal from "../../commons/Modal";
import { fetcher } from "../../commons/utils/client";
import useModal from "../../Hook/useModal";
import Layout from "../../Layout";
import Question from "../Question";
import { Lable, ListHeader, Table, Th, Tr, TrText, ReviewedTag } from "./ListReviews.styled";
import { FaCheckCircle } from "react-icons/fa";
import { IPerformanceModel } from "../Performance";

export interface IReviewModel {
    id: number;
    reviewTo: string;
    reviewBy: string;
    isReview: boolean;
    createdAt: Date;
    updatedAt: Date;
}

const ListReviews: FunctionComponent = () => {
    const { isShowing, toggle, close } = useModal();
    const [review, setReview] = useState<IReviewModel>();
    const token = localStorage.getItem("token") || "";
    const { state } = useContext(UserContext);
    const { data = [], mutate } = useSWR<IReviewModel[]>(
        [REVIEW_LIST_API, token, `username=${state.username}`],
        fetcher
    );

    const closeModal = () => {
        close();
        mutate([...(data as IReviewModel[])]);
    };

    return data && data.length ? (
        <Layout>
            <ListHeader>
                <Lable>List Employees</Lable>
            </ListHeader>
            <Table>
                <Th>Waiting for reviews</Th>
                {data.map((payload: IReviewModel, index: number) => {
                    const onClick = () => {
                        setReview(payload);
                        toggle();
                    };
                    return (
                        <Tr onClick={onClick} key={index}>
                            <TrText>
                                {payload.reviewTo}
                                {payload.isReview ? (
                                    <ReviewedTag>
                                        <FaCheckCircle />
                                    </ReviewedTag>
                                ) : (
                                    ""
                                )}
                            </TrText>
                        </Tr>
                    );
                })}
            </Table>
            <Modal isShowing={isShowing} close={close}>
                <Question review={review as IPerformanceModel} close={closeModal} />
            </Modal>
        </Layout>
    ) : (
        <Layout>
            <Table>
                <Lable>No waiting list</Lable>
            </Table>
        </Layout>
    );
};

export default ListReviews;
