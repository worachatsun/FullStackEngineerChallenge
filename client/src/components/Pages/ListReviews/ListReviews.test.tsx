import React from "react";
import renderer from "react-test-renderer";
import ListReviews from "./index";

describe("<ListReviews />", () => {
    test("should render correctly", () => {
        const render = renderer.create(<ListReviews />).toJSON();
        expect(render).toMatchSnapshot();
    });
});
