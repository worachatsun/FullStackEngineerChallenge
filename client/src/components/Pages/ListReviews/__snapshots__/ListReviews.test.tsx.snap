// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListReviews /> should render correctly 1`] = `
Array [
  <div
    className="sc-AxgMl gyEuQp"
  >
    <div
      className="sc-AxheI dBoGpP"
      onClick={[Function]}
    >
      <img
        alt="logo"
        src="img_logo.png"
      />
    </div>
    <div
      className="sc-Axmtr dLNRmH"
    >
      <div
        className="sc-AxmLO lmWcHq"
      >
        
      </div>
      <button
        className="sc-AxhUy ejktQs"
        onClick={[Function]}
      >
        Logout
      </button>
    </div>
  </div>,
  <div
    className="sc-fzoLsD kuWeDv"
  >
    <div
      className="sc-fzozJi fGxptC"
    >
      <div
        className="sc-fznxsB hQTags"
      >
        <div
          className="sc-fzqNqU joXFMV"
        >
          No waiting list
        </div>
      </div>
    </div>
  </div>,
]
`;
